docs/book/guides/class-based-api/chapter-3.md

```diff
@@ -0,0 +1,231 @@
+
+        # Analyzing the train dataset
+        statistics, schema = analyzer(dataset=train) 
+
+        # Preprocessing the splits
+        train_t, test_t, validation_t = preprocesser( 
+            train_dataset=train,
+            test_dataset=test,
+            validation_dataset=validation,
+            statistics=statistics,
+            schema=schema,
+        )
+
+        # Training the model
+        model = trainer(train_dataset=train_t, validation_dataset=validation_t)
+
+        # Evaluating the trained model
+        evaluator(model=model, dataset=test_t)
+```
+
+You can add your steps to it and run your pipeline as follows:
+
+```python
+# Create the pipeline and run it
+import os
+
+pipeline_instance = TrainingPipeline(
+    datasource=PandasDatasource(PandasDatasourceConfig(path=os.getenv("data"))),
+    splitter=SklearnSplitter(SklearnSplitterConfig(ratios={"train": 0.7, "test": 0.15, "validation": 0.15})),
+    analyzer=PandasAnalyzer(PandasAnalyzerConfig(percentiles=[0.2, 0.4, 0.6, 0.8, 1.0])),
+    preprocesser=SklearnStandardScaler(SklearnStandardScalerConfig(ignore_columns=["has_diabetes"])),
+    trainer=TensorflowBinaryClassifier(TensorflowBinaryClassifierConfig(target_column="has_diabetes")),
+    evaluator=SklearnEvaluator(SklearnEvaluatorConfig(label_class_column="has_diabetes"))
+)
+
+pipeline_instance.run()
+```
+
+Beautiful, now the pipeline is truly doing something. Let's run it!
+
+### Run
+
+You can run this as follows:
+
+```python
+python chapter_3.py
+```
+
+The output will look as follows (note: this is filtered to highlight the most important logs)
+
+```bash
+Creating pipeline: mnist_pipeline
+Cache enabled for pipeline `mnist_pipeline`
+Using orchestrator `local_orchestrator` for pipeline `mnist_pipeline`. Running pipeline..
+Step `importer_mnist` has started.
+Step `importer_mnist` has finished in 1.819s.
+Step `normalize_mnist` has started.
+Step `normalize_mnist` has finished in 2.036s.
+Step `tf_trainer` has started.
+Step `tf_trainer` has finished in 4.723s.
+Step `tf_evaluator` has started.
+`tf_evaluator` has finished in 0.742s.
+```
+
+### Inspect
+
+If you add the following code to fetch the pipeline:
+
+```python
+from zenml.core.repo import Repository
+
+repo = Repository()
+p = repo.get_pipeline(pipeline_name="mnist_pipeline")
+runs = p.runs
+print(f"Pipeline `mnist_pipeline` has {len(runs)} run(s)")
+run = runs[-1]
+print(f"The run you just made has {len(run.steps)} steps.")
+step = run.get_step('evaluator')
+print(
+    f"The `tf_evaluator step` returned an accuracy: {step.output.read()}"
+)
+```
+
+You get the following output:
+
+```bash
+Pipeline `mnist_pipeline` has 1 run(s)
+The first run has 4 steps.
+The `tf_evaluator step` returned an accuracy: 0.9100000262260437
+```
+
+Wow, we just trained our first model! But have not stopped yet. What if did not want to use TensorFlow? Let's swap out our trainers and evaluators for different libraries.
```

Comment:
```suggestion
Wow, we just trained our first model! But we're not done yet. Suppose you don't want to use TensorFlow... Let's swap out our trainers and evaluators for different libraries.
```

Reply:
I have taken out this part as the next stages of the docs is still in progress.